### PR TITLE
[MAINT] use parametrize to test plugins with every answer combo

### PR DESCRIPTION
**Pull request by psobolewskiPhD**
In this PR I try to address the fact that tests failed to catch https://github.com/napari/cookiecutter-napari-plugin/issues/172  
which is to generate and test plugins for each combo of answers, except 4X `n` because that has no modules and no tests.

### CODE REVIEW

File: tests/test_create_template.py [hunk 1]
```diff
@@ -19,20 +19,40 @@ def run_tox(plugin):
...
 
-    run_tox(str(result.project_path))
+    if include_reader_plugin == "y":
+        assert result.project_path.joinpath("src", "foo_bar", "_tests", "test_reader.py").is_file()
+    if include_writer_plugin == "y":
+        assert result.project_path.joinpath("src", "foo_bar", "_tests", "test_writer.py").is_file()
+    if include_sample_data_plugin == "y":
+        assert result.project_path.joinpath("src", "foo_bar", "_tests", "test_sample_data.py").is_file()
+    if include_widget_plugin == "y":
+        assert result.project_path.joinpath("src", "foo_bar", "_tests", "test_widget.py").is_file()
```
Comment (Czaki) [inline tests/test_create_template.py:51]:
```suggestion
    test_path = result.project_path.joinpath("src", "foo_bar", "_tests")
    if include_reader_plugin == "y":
        assert (test_path / "test_reader.py").is_file()
    if include_writer_plugin == "y":
        assert (test_path / "test_writer.py").is_file()
    if include_sample_data_plugin == "y":
        assert (test_path / test_sample_data.py").is_file()
    if include_widget_plugin == "y":
        assert (test_path / "test_widget.py").is_file()
```

File: tests/test_create_template.py [hunk 2]
```diff
@@ -19,20 +19,40 @@ def run_tox(plugin):
...
+
+    # if all are `n` there are no modules or tests    
+    if not (include_reader_plugin == "n" and include_writer_plugin == "n" and include_sample_data_plugin == "n" and include_widget_plugin == "n"):
```
Comment (Czaki) [inline tests/test_create_template.py:54]:
```suggestion
    if "y" in {include_reader_plugin, include_writer_plugin, include_sample_data_plugin, include_widget_plugin}:
```

**Review by Czaki — COMMENTED** [review]
simplify code

**Review by jni — APPROVED** [review]

File: tests/test_create_template.py [hunk 3]
```diff
@@ -19,20 +19,41 @@ def run_tox(plugin):
...
-    assert result.project_path.joinpath("src", "foo_bar", "_tests", "test_reader.py").is_file()
 
-    run_tox(str(result.project_path))
+    test_path = result.project_path.joinpath("src", "foo_bar", "_tests")
+    if include_reader_plugin == "y":
+        assert (test_path / "test_reader.py").is_file()
+    if include_writer_plugin == "y":
+        assert (test_path / "test_writer.py").is_file()
+    if include_sample_data_plugin == "y":
+        assert (test_path / test_sample_data.py").is_file()
```
Comment (Czaki) [inline tests/test_create_template.py:50]:
```suggestion
        assert (test_path / "test_sample_data.py").is_file()
```

**Pull request merged**